src/services/transfer/transfer-out-ehr-core.js

```diff
@@ -18,65 +18,86 @@ export async function transferOutEhrCore({ conversationId, nhsNumber, messageId,
   setCurrentSpanAttributes({ conversationId: conversationId });
   logInfo('EHR transfer out request received');
 
-  let logs = 'EHR has been successfully sent';
-
-  const defaultResult = {
-    hasFailed: false,
-    inProgress: false
-  };
-
   try {
-    const previousTransferOut = await getRegistrationRequestStatusByConversationId(conversationId);
-    if (previousTransferOut !== null) {
-      logInfo(`Duplicate transfer out request`);
-      return {
-        inProgress: true
-      };
+    // Stop transfer if it is a duplicated EHR request
+    if (await isEhrRequestDuplicate(conversationId)) {
+      return;
     }
-    await createRegistrationRequest(conversationId, messageId, nhsNumber, odsCode);
 
-    logInfo('Getting patient health record from EHR repo');
-    const { ehrCore, fragmentMessageIds } = await getEhrCoreAndFragmentIdsFromRepo(
-      nhsNumber,
-      conversationId
-    );
+    await createRegistrationRequest(conversationId, messageId, nhsNumber, odsCode);
 
+    // Stop transfer if ODS codes don't match
     if (!(await patientAndPracticeOdsCodesMatch(nhsNumber, odsCode))) {
-      logs = 'Patients ODS Code in PDS does not match requesting practices ODS Code';
-      await updateConversationStatus(conversationId, Status.INCORRECT_ODS_CODE, logs);
-      return defaultResult;
+      await updateConversationStatus(
+        conversationId,
+        Status.INCORRECT_ODS_CODE,
+        'Patients ODS Code in PDS does not match requesting practices ODS Code'
+      );
+      return;
     }
 
     await updateConversationStatus(conversationId, Status.ODS_VALIDATION_CHECKS_PASSED);
 
-    let { ehrCoreWithUpdatedMessageId, newMessageId } = await updateMessageIdForEhrCore(ehrCore);
-    logInfo(`Successfully replaced message id for ehrCore`);
+    logInfo('Getting patient health record from EHR repo');
 
-    if (fragmentMessageIds?.length > 0) {
-      await createNewMessageIdsForAllFragments(fragmentMessageIds);
-      logInfo(`Successfully created new message id for all fragments`);
-      ehrCoreWithUpdatedMessageId = await updateReferencedFragmentIds(ehrCoreWithUpdatedMessageId);
-      logInfo(`Successfully replaced fragment id reference in ehrCore`);
-    }
+    const { ehrCoreWithUpdatedMessageId, newMessageId } = await getEhrCoreAndUpdateMessageIds(
+      nhsNumber,
+      conversationId
+    );
+    // TODO: Here we should update the message id of this Registration Request in database to newMessageId.
+    // To be addressed in another ticket

```

Comment:
Can we remove this TODO please, if no action has already been taken can we pop this in another ticker for the backlog? :D